sdk/azidentity/managed_identity_client.go

```diff
@@ -173,7 +173,7 @@ func (c *managedIdentityClient) createAuthRequest(ctx context.Context, clientID
 		// need to perform preliminary request to retreive the secret key challenge provided by the HIMDS service
 		key, err := c.getAzureArcSecretKey(ctx, scopes)
 		if err != nil {
-			return nil, &AuthenticationFailedError{inner: err, msg: "Failed to retreive secret key from the identity endpoint."}
+			return nil, &AuthenticationFailedError{msg: "Failed to retreive secret key from the identity endpoint."}
```

Comment:
Nit: Idiom for error messages is to not start with a capital letter and not end with punctuation (the linter will flag this, but likely can't see through this case).